### DISCUSSION TITLE
Thumbnails for any custom file?

### DISCUSSION THREAD

**Discussion by Davey-101** [Xanashi/Icaros#127] [Feature Requests]
Would it be possible to create thumbnails for any custom file type, for example "file.type", perhaps by putting image files in the same folder with names like "file.type.jpg"?

**Comment by Xanashi** [comment]
In theory, it would be possible if you created your custom format as a zip file, that includes an image file in the top directory. 

It's not possible for thumbnail shell extensions to find other files in the same directory as the source file, simply because Windows doesn't
give the shell extension that information. It only provides a raw data stream of the file, that it wants thumbnailed.

**Reply by Davey-101** [reply]
Thanks!

What if the image files were stored in a single folder so the shell extension knows where to look?

**Reply by Xanashi** [reply]
It may be possible, but it's not a recommended design. The problem is, if a shell extension starts pulling files from other places in the system, it could introduce difficult to debug bugs, and potentially open up for vulnerabilities that could be exploited. 
I'm not too keen on this approach.

**Comment by clsid2** [comment]
You can use `folder.jpg` to give a directory a thumbnail. That is a Windows feature and also works without Icaros.

**Reply by Davey-101** [reply]
This was almost the solution. The problem is that I have lots of files that I would like to search and filter with saved searches and I need the results sorted by date modified. 

I could change the searches to return folders and not files and that would solve the thumbnail problem but unfortunately Windows appears to not update the date of a folder if the contents are modified. The folder date only changes if a file is created or deleted in the folder.

Thanks anyway.

**Reply by clsid2** [reply]
If you create `file.type.jpg` with desired thumb for each file, then you can search for `*.type.jpg`

**Reply by Davey-101** [reply]
Thanks but I don't see how that helps me. I need the search result to have the modified date of the data file and I need some way to open the file itself from the search result. 

Opening a folder and clicking the one data file inside it would work, if only the folder modified date would update properly. If I do go down this route, I might use another 3rd party tool to update the folder date like this one https://www.nirsoft.net/utils/folder_time_update.html.

**Reply by Xanashi** [reply]
Depending on how you work with your data file, creating a custom zip based filetype could still be a viable solution.
If you zip you data + thumbnail in a file and rename it to file.type, then add the .type to Icaros, you would get a thumbnail, for that file.

You then need to make sure .type opens with something like 7zip, and if you open your data file directly in 7zip's GUI, and modify it,
7zip will automatically ask you, if you wish to save the changes to your archive, and this will update your .type file. 

This will also update the last modified date of the .type file, which can then be used when searching.
